samza-azure/src/main/java/org/apache/samza/system/azureblob/avro/AzureBlobOutputStream.java

```diff
@@ -335,6 +335,11 @@ public void run() {
             // StageBlock generates exception on Failure.
             stageBlock(blockIdEncoded, outputStream, blockSize);
             break;
+          } catch (InterruptedException e) {
+            String msg = "Upload block for blob: " + blobAsyncClient.getBlobUrl().toString()
+                + " failed for blockid: " + blockId + " due to InterruptedException ";
+            LOG.error(msg, e);
+            throw new AzureException("InterruptedException encountered during block upload. Will not retry.", e);
```

Comment:
Is there a reason why on a thread being interrupted should it not be retried?

Reply:
original idea of retry was to deal with azure blob storage connection issues. 

Now the user of AzureBlobSystemProducer has a need where they want to interrupt the thread themselves if it doesnt stop within a set period of time (stop on thread invoked by them again). In such a scenario, they expect the thread to be killed immediately and producer bubble up the exception so that they can handle the exception in an appropriate manner. since they are dealing at the thread level and not producer level, it becomes hard for them to identify the producer of the thread and handle producer exceptions.